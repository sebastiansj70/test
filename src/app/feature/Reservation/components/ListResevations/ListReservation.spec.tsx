import '@testing-library/jest-dom';
import React from 'react';
import { RenderResult, render, wait, fireEvent } from '@testing-library/react';
import { ReservationList } from '.';
import { setTextEvent } from 'app/shared/utils/test';

describe('', () => {
  let componentWrapper: RenderResult;
  let componentProps: React.ComponentProps<typeof ReservationList>;

  beforeEach(() => {
    componentProps = {
      reservationLists: [
        {
          idTicket: 1,
          telefonoUsuario: 3186905006,
          nombreUsuario: 'juan',
          horaIngreso: 1626283800000,
          horaSalida: 1626283800000,
          idCancha: 1,
          valor: 50000,
        },
        {
          idTicket: 2,
          telefonoUsuario: 3186905006,
          nombreUsuario: 'Sebastian',
          horaIngreso: 1626283800000,
          horaSalida: 1626283800000,
          idCancha: 2,
          valor: 50000,
        },
      ],
      reservation: {
        idTicket: 3,
        telefonoUsuario: 3186905006,
        nombreUsuario: 'juan',
        horaIngreso: 1626283800000,
        horaSalida: 1626283800000,
        idCancha: 1,
        valor: 50000,
      },
      saveReservation: jest.fn(),
      updateReservation: jest.fn(),
    };

    componentWrapper = render(<ReservationList {...componentProps} />);
  });

  it('should match snapshot', () => {
    expect(componentWrapper.container).toMatchSnapshot();
  });

  it('Deber tener un titulo', () => {
    const elem = componentWrapper.container;
    const tittle = elem.querySelectorAll('h1');

    expect(tittle[0].textContent).toMatch(/reservaciones/i);
  });
});
