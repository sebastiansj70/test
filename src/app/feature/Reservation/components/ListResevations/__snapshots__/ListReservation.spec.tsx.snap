// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="sc-AxheI dtDSCr"
  >
    <h1
      class="sc-fzozJi jIbZRk"
    >
      Lista de reservaciones
    </h1>
    <div
      id="reservationsList"
    >
      <div
        class="sc-AxjAm bWaaKr"
      >
        <div
          class="sc-AxiKw iFfnQU"
        >
          <img
            class="sc-AxhUy jHSZEE"
            src="logo-ball.png"
          />
          <h1>
            San Juan
          </h1>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Teléfono:  
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            3186905006
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Nombre : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            juan
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Ingreso : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            1626283800000
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Salida :
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            1626283800000
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Cancha : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            1
          </h3>
        </div>
        <h5
          class="sc-AxgMl SfUKN"
        >
          --------------------------------------------------------
        </h5>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            valor : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            50000
          </h3>
        </div>
      </div>
    </div>
    <div
      id="reservationsList"
    >
      <div
        class="sc-AxjAm bWaaKr"
      >
        <div
          class="sc-AxiKw iFfnQU"
        >
          <img
            class="sc-AxhUy jHSZEE"
            src="logo-ball.png"
          />
          <h1>
            San Juan
          </h1>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Teléfono:  
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            3186905006
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Nombre : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            Sebastian
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Ingreso : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            1626283800000
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Salida :
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            1626283800000
          </h3>
        </div>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            Cancha : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            2
          </h3>
        </div>
        <h5
          class="sc-AxgMl SfUKN"
        >
          --------------------------------------------------------
        </h5>
        <div
          class="sc-AxirZ jRMgv"
        >
          <h3>
            valor : 
          </h3>
          <h3
            class="sc-AxhCb izLcMB"
          >
            50000
          </h3>
        </div>
      </div>
    </div>
    <div />
  </div>
</div>
`;
